crates/cxx-qt-lib/src/lib.rs

```diff
@@ -8,8 +8,19 @@
 compile_error!("qt_version_major must be either \"5\" or \"6\"");
 
 mod core;
+use std::cmp::Ordering;
+
 pub use crate::core::*;
 
+fn get_ordering(ord: i8) -> Option<Ordering> {
+    match ord {
+        -1 => Some(Ordering::Less),
+        0 => Some(Ordering::Equal),
+        1 => Some(Ordering::Greater),
+        _ => None,
+    }
```

Comment:
Is there a way to register `std::cmp::Ordering` for C++ so that we can avoid this conversion?
I know **cxx** has `extern "Rust"`, but I wasn't able to get it working to register this type from the STL.

Reply:
Maybe use a Shared Enum https://cxx.rs/shared.html then impl From for it do that you can go .into() from the CXX / C++ enum to `std::cmp::Ordering` ?

Reply:
But if I understand correctly shared enums only work for user-defined enums. I don't see any way to register the already existing `std::cmp::Ordering` or am I missing something?

Reply:
Right, i can't see a sane way to register an existing Rust enum to C++, it'd have to be an opaque type.

So you would have to create a second shared enum in the bridge

```rust
enum CXXOrdering { Less, Equal, Greater }
```

Then do like `impl From<CXXOrdering> for Order { ... }`.

However this would need to be done somewhere common to avoid creating one in each bridge. So maybe just returning a number and using a helper method is fine for now :-)